### PR TITLE
#9Task. valery.aleshka

### PR DESCRIPTION
**Pull request by Valeryaleshka**
https://travis-ci.org/github/Valeryaleshka/js-assignments/builds/715789878

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -26,7 +26,12 @@
  *
  */
 function getComposition(f,g) {
-    throw new Error('Not implemented');
+
+    return function(x) {
+        return f(g(x));
+      };
+
+    //throw new Error('Not implemented');
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: It is better to delete unnecessary comments

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -65,7 +75,55 @@ function getPowerFunction(exponent) {
...
+                c = arguments[0];
+            }
+
+
+        }
+    }
+
+    return function (x) {
+        let y = a * Math.pow(x,a) + b * x + c;
+        if(y===0){
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: It is necessary to respect the indentation between operators

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -84,7 +142,18 @@ function getPolynom() {
  *   memoizer() => the same random number  (next run, returns the previous cached result)
  */
 function memoize(func) {
-    throw new Error('Not implemented');
+
+    let memory = null;
+    return function(arg){
+        if(memory===null){
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: Must be indented after conditional statements "if", between operators "====", before brackets "function(arg) {"

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -171,7 +283,19 @@ function partialUsingArguments(fn) {
  *   getId10() => 11
  */
 function getIdGeneratorFunction(startFrom) {
-    throw new Error('Not implemented');
+
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js:271]: Unnecessary blank lines should be removed

File: task/09-functions-n-closures-tasks.js [hunk 5]
```diff
@@ -171,7 +283,19 @@ function partialUsingArguments(fn) {
  *   getId10() => 11
  */
 function getIdGeneratorFunction(startFrom) {
-    throw new Error('Not implemented');
+    
+    let id = startFrom;
+    let bool = false;
+    return function(){
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: Here you can simply return function with increment "return () => startFrom++"

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by Valeryaleshka** [comment]
I've fixed everything.
New Travis-ci build.
https://travis-ci.org/github/Valeryaleshka/js-assignments/builds/721848702

**Review by KostyaKov — COMMENTED** [review]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**